PR modifi controller

MatvijVanij opened the pull request.

Commit:
modifi controller

Commit:
fixup! modifi controller

SemBen commented on DataAccess.InsightDatabase/Repositories/CourseRepository.cs:
```diff
@@ -11,19 +11,19 @@ namespace DataAccess.InsightDatabase.Repositories
     public class CourseRepository : ICourseRepository
     {
         public IDbConnection DBConnection { get; }
+        private ICourseRepository _courseRepository; 
```
private readonly

SemBen commented on DataAccess.InsightDatabase/Repositories/GroupWithStudentRepository.cs:
```diff
@@ -10,19 +10,19 @@ namespace DataAccess.InsightDatabase.Repositories
     public class GroupWithStudentRepository : IGroupWithStudentRepository
     {
         public IDbConnection DBConnection { get; }
+        private IGroupWithStudentRepository _groupWithStudentRepository;
```
private readonly

SemBen commented on DataAccess.InsightDatabase/Repositories/HomeworkRepository.cs:
```diff
@@ -12,19 +12,19 @@ public class HomeworkRepository : IHomeworkRepository
     {
 
         public IDbConnection DBConnection { get; }
+        private IHomeworkRepository _homeworkRepository;
```
private readonly

SemBen commented on DataAccess.InsightDatabase/Repositories/HomeworkRepository.cs:
```diff
@@ -12,19 +12,19 @@ public class HomeworkRepository : IHomeworkRepository
         }
 
-        public async Task<bool> CreateHomeworkAsync(Homework homework)
+        public async Task<bool> CreateHomeworkWithinLessonAsync(Homework homework,Guid LessonID)
```
сначала айди, потом обьект

SemBen commented on DataAccess.InsightDatabase/Repositories/HomeworkRepository.cs:
```diff
@@ -12,19 +12,19 @@ public class HomeworkRepository : IHomeworkRepository
-                IHomeworkRepository homeworkRepository = DBConnection.As<IHomeworkRepository>();
-
-                return await homeworkRepository.CreateHomeworkAsync(homework);
+                return await _homeworkRepository.CreateHomeworkWithinLessonAsync(homework,LessonID);
```
сначала айди

SemBen commented on DataAccess.InsightDatabase/Repositories/LessonRepository.cs:
```diff
@@ -11,19 +11,19 @@ namespace DataAccess.InsightDatabase.Repositories
     public class LessonRepository : ILessonRepository
     {
         public IDbConnection DBConnection { get; }
+        private ILessonRepository _lessonRepository;
```
private readonly

SemBen commented on DataAccess.InsightDatabase/Repositories/UserDetailRepository.cs:
```diff
@@ -2,6 +2,8 @@
 using System.Data;
 using System.Collections.Generic;
+using System.Data;
+using System.Text;
```
удали лишние юзинги

SemBen commented on DataAccess.InsightDatabase/Repositories/UserDetailRepository.cs:
```diff
@@ -13,62 +15,39 @@ namespace DataAccess.InsightDatabase.Repositories
     public class UserDetailRepository : IUserDetailRepository
     {
         public IDbConnection DBConnection { get; }
+        private IUserDetailRepository _userDetailRepository;
```
private readonly

SemBen commented on DataAccess.InsightDatabase/Repositories/UserDetailRepository.cs:
```diff
@@ -13,62 +15,39 @@ namespace DataAccess.InsightDatabase.Repositories
-        public async Task<IEnumerable<UserDetail>> GetAllUsersDetailAsync()
+        public async Task<UserDetail> GetUserDetailByIDAsync(Guid id)
         {
             IUserDetailRepository userDetailRepository = DBConnection.As<IUserDetailRepository>();
```
удали строку

SemBen commented on DataAccess.InsightDatabase/Repositories/UserWithRoleRepository.cs:
```diff
@@ -13,63 +13,19 @@ namespace DataAccess.InsightDatabase.Repositories
     public class UserWithRoleRepository : IUserWithRoleRepository
     {
         public IDbConnection DBConnection { get; }
+        private IUserWithRoleRepository _userWithRoleRepository;
```
private readonly

SemBen commented on Domain/Interfaces/HomeworkRepositoryInterfaces/IHomeworkRepository.cs:
```diff
@@ -8,9 +8,9 @@ namespace Domain.Interfaces.HomeworkRepositoryInterfaces
-        Task<bool> CreateHomeworkAsync(Homework homework);
+        Task<IEnumerable<Homework>> GetAllHomeworkByCourseIDAsync(Guid CourseID);
+        Task<Homework> GetHomeworkByLessonIdAsync(Guid LessonID);
+        Task<bool> CreateHomeworkWithinLessonAsync(Homework homework, Guid LessonID);
```
айди сначала

SemBen commented on DataAccess.InsightDatabase/Repositories/LessonRepository.cs:
```diff
@@ -48,15 +46,15 @@ public async Task<Lesson> GetLessonByIdAsync(Guid id)
         }
 
-        public async Task<bool> CreateLessonAsync(Lesson lesson)
+        public async Task<bool> CreateLessonWithinCourseAsync(Lesson lesson,Guid CoursID)
```
сначала айди

SemBen commented on Domain/Interfaces/IDBContext.cs:
```diff
@@ -5,6 +5,8 @@
 using Domain.Interfaces.UserRepositoryInterfaces;
 ﻿using Domain.Interfaces.CourseRepositoryIntarfaces;
+using Domain.Interfaces.UserRepositoryInterfaces;
+using Domain.Interfaces.UserRepositoryInterfaces;
```
удали лишние юзинги

SemBen commented on Domain/Interfaces/IDBContext.cs:
```diff
@@ -17,16 +19,15 @@ public interface IDBContext
         IHomeworkRepository HomeworkRepository { get; }
-        ICourseProgramRepository CourseProgramRepository { get; }
         ICourseRepository CourseRepository { get; }
-        ILessonRepository LessonRepository { get; }
```
уверен что нужно удалять урок?

SemBen commented on Domain/Interfaces/IDBContext.cs:
```diff
@@ -17,16 +19,15 @@ public interface IDBContext
         ITeacherRepository TeacherRepository { get; }
         IGroupRepository GroupRepository { get; }
         ISolutionRepository SolutionRepository { get; }
 
```
удали пустые строки

SemBen commented on Domain/Interfaces/LessonRepositoryInterfaces/ILessonRepository.cs:
```diff
@@ -13,12 +13,12 @@ public interface ILessonRepository : IRepository
         //TODO one to many
         Task<Lesson> GetLessonByIdAsync(Guid id);
-        Task<bool> CreateLessonAsync(Lesson lesson);
+        Task<bool> CreateLessonWithinCourseAsync(Lesson lesson, Guid CoursID);
```
сначала айди

SemBen commented on Domain/Interfaces/UserRepositoryInterfaces/IUserWithRoleRepository.cs:
```diff
@@ -10,16 +10,12 @@ namespace Domain.Interfaces.UserRepositoryInterfaces
 
         [Recordset(typeof(UserWithRole), typeof(Role))]
         Task<UserWithRole> GetUserWithRoleByLoginAndPasswordAsync(string login, string password);
 
```
удали пустые строки

SemBen commented on WebApi/Controllers/HomeworkController.cs:
```diff
@@ -18,22 +18,22 @@ public HomeworkController(IDBContext dbContext)
-        [HttpGet]
-        public async Task<IEnumerable<Homework>> GetHomeworks()
+        [HttpGet("{courseid}")]
+        public async Task<IEnumerable<Homework>> GetAllHomeworkByCourseIDAsync(Guid CourseID)
```
имя метода без Async

SemBen commented on WebApi/Controllers/HomeworkController.cs:
```diff
@@ -18,22 +18,22 @@ public HomeworkController(IDBContext dbContext)
 
-        [HttpGet("id")]
-        public async Task<Homework> GetHomeworkById(Guid id)
+        [HttpGet("lessonid")]
```
возьму лесонайди в фигурные скобки

sinerger replied:
нет, не нужно

SemBen commented on WebApi/Controllers/HomeworkController.cs:
```diff
@@ -18,22 +18,22 @@ public HomeworkController(IDBContext dbContext)
-        [HttpGet("id")]
-        public async Task<Homework> GetHomeworkById(Guid id)
+        [HttpGet("lessonid")]
+        public async Task<Homework> GetHomeworkByLessonIdAsync(Guid LessonID)
```
имя метода без Async

SemBen commented on WebApi/Controllers/HomeworkController.cs:
```diff
@@ -18,22 +18,22 @@ public HomeworkController(IDBContext dbContext)
 
         [HttpPost]
-        public async Task<bool> GreateHomework(Homework homework)
+        public async Task<bool> CreateHomeworkWithinLessonAsync(Homework homework, Guid LessonID)
```
имя метода без Async

SemBen replied:
айди сначала

SemBen commented on WebApi/Controllers/HomeworkController.cs:
```diff
@@ -18,22 +18,22 @@ public HomeworkController(IDBContext dbContext)
+        public async Task<bool> CreateHomeworkWithinLessonAsync(Homework homework, Guid LessonID)
         {
-            return await _dbContext.HomeworkRepository.CreateHomeworkAsync(homework);
+            return await _dbContext.HomeworkRepository.CreateHomeworkWithinLessonAsync(homework,LessonID);
```
айди сначала

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
 
         [HttpGet]
-        public async Task<IEnumerable<Lesson>> GetLessons()
+        public async Task<IEnumerable<Lesson>> GetAllLessonsAsync()
```
имя метода без Async

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
-        [HttpGet("id")]
-        public async Task<Lesson> GetLessonById(Guid id)
+        [HttpGet("{id}")]
+        public async Task<Lesson> GetLessonByIdAsync(Guid id)
```
имя метода без Async

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
-        [HttpDelete("id")]
-        public async Task<bool> DeleteLesson(Guid id)
+        [HttpDelete("{id}")]
+        public async Task<bool> DeleteLessonAsync(Guid id)
```
имя метода без Async

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
 
         [HttpPost]
-        public async Task<bool> CreateLesson(Lesson lesson)
+        public async Task<bool> CreateLessonWithinCourseAsync(Lesson lesson,Guid courseID)
```
айди сначала

SemBen replied:
имя метода без Async

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
+        public async Task<bool> CreateLessonWithinCourseAsync(Lesson lesson,Guid courseID)
         {
-            return await _dbContext.LessonRepository.CreateLessonAsync(lesson);
+            return await _dbContext.LessonRepository.CreateLessonWithinCourseAsync(lesson, courseID);
```
айди сначала

SemBen commented on WebApi/Controllers/LessonController.cs:
```diff
@@ -19,31 +19,31 @@ public LessonController(IDBContext dBContext)
 
         [HttpPut]
-        public async Task<bool> UpdateLesson(Lesson lesson)
+        public async Task<bool> UpdateLessonAsync(Lesson lesson)
```
имя метода без Async

SemBen commented on WebApi/Controllers/UserDetailController.cs:
```diff
@@ -21,12 +21,6 @@ public UserDetailController(IDBContext dbContext)
-        }
-        
         [HttpGet("{id}")]
         public async Task<UserDetail> GetUserDetailByID (Guid id)
```
имя метода без Async

SemBen commented on WebApi/Controllers/UserDetailController.cs:
```diff
@@ -40,7 +34,7 @@ public async Task<bool> CreateDetailInfoForUserAsync(UserDetail user)
 
         [HttpPut]
-        public async Task<bool> UpdateUserDetail(UserDetail user)
+        public async Task<bool> UpdateDetailInfoForUserAsync(UserDetail user)
```
имя метода без Async

SemBen commented on WebApi/Controllers/UserWithRoleController.cs:
```diff
@@ -1,6 +1,9 @@
+using Domain.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
```
удали лишние юзинги и отсортируй

SemBen commented on WebApi/Controllers/UserWithRoleController.cs:
```diff
@@ -20,12 +23,6 @@ public UserWithRoleController(IDBContext dbContext)
-            return await _dbContext.UserWithRoleRepository.GetUsersWithRoleAsync();
-        }
-
         [HttpGet("id")]
```
возьми айди в фигурные скобки

SemBen commented on WebApi/Controllers/UserWithRoleController.cs:
```diff
@@ -38,12 +35,6 @@ public async Task<UserWithRole> GetUserWithRoleByLoginAndPassword(string login,
-            return await _dbContext.UserWithRoleRepository.CreateUserWithRoleAsync(user);
-        }
-
         [HttpDelete("id")]
```
возьми айди в фигурные скобки

SemBen reviewed (commented):
Подправь

sinerger commented on DataAccess.InsightDatabase/DBContext.cs:
```diff
@@ -2,22 +2,26 @@
 {
     public class DBContext : IDBContext
     {
+        public DBContext(IDbConnection dbConnection)
```
Конструктор идет после полей и после свойств

Commit:
fixup! fixup! modifi controller

fixup! fixup! fixup! modifi controller

fixup! fixup! fixup! fixup! modifi controller

MatvijVanij committed:
Merge branch 'dev' into ModifyController
